plugin/cmd/kube-scheduler/app/configurator.go

```diff
@@ -186,3 +191,113 @@ func (sc schedulerConfigurator) Create() (*scheduler.Config, error) {
+func verifyNewPolicyConfig(obj interface{}) error {
+	newConfig, ok := obj.(*v1.ConfigMap)
+	if !ok {
+		return fmt.Errorf("cannot convert obj to *v1.ConfigMap: %v", obj)
+	}
+	newPolicy, ok := newConfig.Data[options.SchedulerPolicyConfigMapKey]
+	if !ok {
+		return fmt.Errorf("No element with key = '%v' is found in the ConfigMap.Data", options.SchedulerPolicyConfigMapKey)
+	}
+	configData := []byte(newPolicy)
+	var policy schedulerapi.Policy
+	if err := runtime.DecodeInto(latestschedulerapi.Codec, configData, &policy); err != nil {
+		return fmt.Errorf("invalid scheduler policy configuration: %v", err)
+	}
+	// validate the policy configuration
+	if err := validation.ValidatePolicy(policy); err != nil {
+		return fmt.Errorf("The new scheduler policy is invalid. Will keep using the old policy! Error: %v", err)
+	}
+	return nil
+}
+
+func (sc *schedulerConfigurator) addPolicyConfigMap(obj interface{}) {
+	glog.Info("Received a request to add a scheduler policy config.")
+	err := verifyNewPolicyConfig(obj)
+	if err != nil {
+		glog.Error(err)
+		return
+	}
+	// If things are in order, kill the scheduler to apply the new config.
+	sc.KillScheduler()
+}
+
+func (sc *schedulerConfigurator) updatePolicyConfigMap(oldObj, newObj interface{}) {
+	glog.Info("Received an update to the scheduler policy config.")
+	_, ok := oldObj.(*v1.ConfigMap)
+	if !ok {
+		glog.Errorf("cannot convert oldObj to *v1.ConfigMap: %v", oldObj)
+		return
+	}
+	err := verifyNewPolicyConfig(newObj)
+	if err != nil {
+		glog.Error(err)
+		return
+	}
+	// If things are in order, kill the scheduler to apply the new config.
+	sc.KillScheduler()
+}
+
+
+func (sc *schedulerConfigurator) deletePolicyConfigMap(obj interface{}) {
+	glog.Info("Scheduler's policy ConfigMap is deleted.")
+	switch t := obj.(type) {
+	case *v1.ConfigMap:  // Nothing is needed. Jump out of the switch.
+	case cache.DeletedFinalStateUnknown:
+		_, ok := t.Obj.(*v1.ConfigMap)
+		if !ok {
+			glog.Errorf("cannot convert to *v1.ConfigMap: %v", t.Obj)
+			return
+		}
+	default:
+		glog.Errorf("cannot convert to *v1.ConfigMap: %v", t)
+		return
+	}
+	sc.KillScheduler()
+}
+
+// schedulerKillFunc is a function that kills the scheduler. It is here mainly for testability. Tests set it to a function to perform an action that can be verified in tests instead of the default behavior which causes the scheduler to die.
+var SchedulerKillFunc func() = nil
+
+func (sc *schedulerConfigurator) KillScheduler() {
+	if SchedulerKillFunc != nil {
+		SchedulerKillFunc()
+	} else {
+		if sc.schedulerConfig != nil {
+			glog.Infof("Scheduler is going to die (and restarted) in order to update its policy.")
+			close(sc.schedulerConfig.StopEverything)
+			// The sleep is only to allow cleanups to happen.
+			time.Sleep(2 * time.Second)
+			glog.Flush()
+			os.Exit(0)
+		}
+		glog.Infof("Scheduler is not going to exit, as it doesn't seem to be initialized yet.")
```

Comment:
I'm curious about this case. Did you observe this in reality? Would it be better to just exit in this case too?